Misc/NEWS.d/next/Library/2020-05-13-15-32-13.bpo-40607.uSPFCi.rst

```diff
@@ -0,0 +1,2 @@
+When cancelling the task due to a timeout, :meth:`asyncio.wait_for` will now
+reraise exceptions triggered by custom cancellation handling.
```

Comment:
How about this wording?

> When cancelling a task due to timeout, :meth:`asyncio.wait_for` will now raise the exception if an error happens during cancellation.

I think it would also be good to update the `wait_for()` docs with the new behavior.

Reply:
I am not sure about your rewording:
- `asyncio.wait_for` already raises an exception (`TimeoutError`) if an error happens during cancellations
- it can misunderstood as it will raise some "special" exception, something like `CancellationFailedException`.

My idea is outline that same exception is propagated. I think next version is more accurate:
> When cancelling a task due to timeout, :meth:asyncio.wait_for will now **propagate** the exception if an error happens during cancellation.